### PR TITLE
Updated MLJ Interface with predict function

### PR DESCRIPTION
**Pull request by PyDataBlog**

### CODE REVIEW

File: src/mlj_interface.jl [hunk 1]
```diff
@@ -94,15 +94,20 @@ function MMI.fit(m::KMeans, verbosity::Int, X)
...
-              converged=fitresult.converged, totalcost=fitresult.totalcost,
-              labels=fitresult.assignments)
+
+    report = (cluster_centers=result.centers, iterations=result.iterations,
+              converged=result.converged, totalcost=result.totalcost,
+              assignments=result.assignments, labels=cluster_labels)
+
+
     """
     # TODO: warn users about non convergence
```
Comment (Arkoniak) [inline src/mlj_interface.jl:111]: Shouldn't it be removed? It looks like next line do necessary warning.

Comment (PyDataBlog) [reply src/mlj_interface.jl:111]: Yes, you are right. I will remove it now.

File: src/mlj_interface.jl [hunk 2]
```diff
@@ -94,15 +94,20 @@ function MMI.fit(m::KMeans, verbosity::Int, X)
...
     # fit model and get results
     verbose = verbosity > 0  # Display fitting operations if verbosity > 0
-    fitresult = ParallelKMeans.kmeans(algo, DMatrix, m.k;
+    result = ParallelKMeans.kmeans(algo, DMatrix, m.k;
                                       n_threads = m.threads, k_init=m.k_init,
                                       max_iters=m.max_iters, tol=m.tol, init=m.init,
                                       verbose=verbose)
 
+    cluster_labels = MMI.categorical(1:m.k)
+    fitresult = (result.centers, cluster_labels, result.converged)
```
Comment (Arkoniak) [inline src/mlj_interface.jl]: I think we can go with `NamedTuples`
```
fitresult = (centers = result.centers, labels = cluster_labels, converged = result.converged)
```

It should not affect those who want to use indices, but will be much more convenient to manipulate.

File: src/mlj_interface.jl [hunk 3]
```diff
@@ -143,21 +140,37 @@ function MMI.transform(m::KMeans, fitresult, Xnew)
     end
 
     # Warn users if fitresult is from a `non-converged` fit
-    if !fitresult[end].converged
+    if !(fitresult[end])
```
Comment (Arkoniak) [inline src/mlj_interface.jl]: With named tuples it'll be `fitresult.converged`

File: src/mlj_interface.jl [hunk 4]
```diff
@@ -143,21 +140,37 @@ function MMI.transform(m::KMeans, fitresult, Xnew)
...
         @warn "Failed to converge. Using last assignments to make transformations."
     end
 
-    # results from fitted model
-    results = fitresult[1]
-
     # use centroid matrix to assign clusters for new data
-    centroids = results.centers
+    centroids = fitresult[1]
```
Comment (Arkoniak) [inline src/mlj_interface.jl]: `fitresult.centers`

**Comment by PyDataBlog** [comment]
Updated based on feedback.

File: src/mlj_interface.jl [hunk 5]
```diff
@@ -143,21 +139,36 @@ function MMI.transform(m::KMeans, fitresult, Xnew)
     end
 
     # Warn users if fitresult is from a `non-converged` fit
-    if !fitresult[end].converged
+    if !(fitresult.converged)
```
Comment (Arkoniak) [inline src/mlj_interface.jl]: You do not need parenthesis here.

File: src/mlj_interface.jl [hunk 6]
```diff
@@ -143,21 +139,36 @@ function MMI.transform(m::KMeans, fitresult, Xnew)
...
 
-    # results from fitted model
-    results = fitresult[1]
-
     # use centroid matrix to assign clusters for new data
-    centroids = results.centers
-    distances = Distances.pairwise(Distances.SqEuclidean(), DMatrix, centroids; dims=2)
-    preds = argmin.(eachrow(distances))
-    return MMI.table(reshape(preds, :, 1), prototype=Xnew)
+    distances = Distances.pairwise(Distances.SqEuclidean(), DMatrix, fitresult.centers; dims=2)
```
Comment (Arkoniak) [inline src/mlj_interface.jl:147]: I am not sure about `SqEuclidean` here. 
Despite the fact that internally most of the time we operate with the squared norm (for performance), our final clusters are calculated in the usual `Euclidean` metric.

I think, we should switch here.

Comment (PyDataBlog) [reply src/mlj_interface.jl:147]: Switching.

**Pull request merged**